02-oop-inheritance/keypoints/2_1-6/script.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Created by MaximilianoDaniel on 15/09/2015.
+ * Exercises done
+ * 1- Create a Movie object
+ * 2- Instantiate some of your favorite movies and play with them in the console.
+ * 3- Add a MovieObserver class that listen for "playing" and "sttoped" events
+ * 4- Publish "playing" event on Movie.play()
+ * 5- Publish "stopped" event on Movie.stop()
+ * 6- Log to console when each event is fired
+ * */
+var inheritPrototype = function (childClass, parentClass){
+
+    var copyOfParent = Object.create(parentClass.prototype);
+
+    //Is necessary to point to the childClass constructor
+    copyOfParent.constructor = childClass;
+    childClass.prototype = copyOfParent;
+};
+
+//MovieObserver can listen one or more actions like play or stop
+var MovieObserver = function (listenTo) {
+    this.listenTo = listenTo || [] ;
+};
+
+MovieObserver.prototype.play = function (movie) {
+    console.log('Playing: ' + movie.title);
+};
+
+MovieObserver.prototype.stop = function (movie) {
+    console.log('Stopped: ' + movie.title);
+};
+
+//The observerList can publish the actions in your own list of observer
+var ObserverList = function () {
+  this.observerList = [];
+};
+
+ObserverList.prototype.publish = function (action) {
+    var observerIndex = 0;
+
+    for (observerIndex ; observerIndex < this.observerList.length ; observerIndex++) {
+        for (var listenToIndex = 0 ; listenToIndex < this.observerList[observerIndex].listenTo.length ; listenToIndex++) {
+            if (this.observerList[observerIndex].listenTo[listenToIndex] === action) {
+                    if (action === 'play') {
+                        this.observerList[observerIndex].play(this);
+                    } else if (action === 'stop') {
+                        this.observerList[observerIndex].stop(this);
+                    }
+            }
+        }
+    }
+};
+
+ObserverList.prototype.subscribe = function (observer) {
+    this.observerList.push(observer);
+};
+
+ObserverList.prototype.unsubscribe = function (observer) {
+    var observerIndex = 0;
+
+    for (observerIndex ; observerIndex < this.observerList.length ; observerIndex++) {
+        if (this.observerList[observerIndex] === observer) {
+            this.observerList.slice(observerIndex, 1);
```

Comment:
If you want to remove an item from ``observerList`` array, are you sure ``slice`` is the method you're looking for? :P